src/ng/filter/orderBy.js

```diff
@@ -74,11 +75,13 @@
  *
  * The default, built-in comparator should be sufficient for most usecases. In short, it compares
  * numbers numerically, strings alphabetically (and case-insensitively), for objects falls back to
- * using their index in the original collection, and sorts values of different types by type.
+ * using their index in the original collection, sorts values of different types by type and puts `undefined` and `null` values at the end of the sorted list.
  *
  * More specifically, it follows these steps to determine the relative order of items:
  *
- * 1. If the compared values are of different types, compare the types themselves alphabetically.
+ * 1. If the compared values are of different types:
+ *    - Explicitly check for `null` and `undefined` values and treat them bigger as any other value.
+ *    - If both values are different from `null` and `undefined`, compare the types themselves alphabetically.
```

Comment:
I would change the bullets to make the relative order of `null` vs `undefined` more explicit:

> - If one of the values is `undefined`, consider it "greater than" the other.
> - Else if one of the values is `null`, consider it "greater than" the other.
> - Else compare the types themselves aplhabetically.